tests/frontend/onnx/test_context.py

```diff
@@ -0,0 +1,82 @@
+import onnx
+import pytest
+
+try:
+    from onnx import TensorProto, ValueInfoProto
+
+    from xdsl.frontend.onnx.context import (
+        OnnxXdslMapping,
+        visit_value_info,
+    )
+except ImportError as exc:
+    print(exc)
+    pytest.skip("onnx is an optional dependency", allow_module_level=True)
+
+
+def test_visit_value_info():
+    # initialize context
+    ctx = OnnxXdslMapping()
+
+    # create ValueInfoProto input tensor
+    input_value_info = ValueInfoProto()
+    input_value_info.name = "input_tensor"
+
+    # define the type of the input tensor
+    input_tensor_type = input_value_info.type.tensor_type
+    input_tensor_type.elem_type = TensorProto.FLOAT
+    input_tensor_type.shape.dim.extend(
+        [
+            onnx.TensorShapeProto.Dimension(dim_value=1),
+            onnx.TensorShapeProto.Dimension(dim_value=3),
+            onnx.TensorShapeProto.Dimension(dim_value=224),
+            onnx.TensorShapeProto.Dimension(dim_value=224),
+        ]
+    )
+
+    # run visit_value_info with empty context
+    visit_value_info(input_value_info, ctx)
+
+    # check keys in context
+    keys = list(ctx.type_by_name.keys())
+    assert keys == ["input_tensor"]
+
+    # check type info
+    type_info = str(ctx.type_by_name["input_tensor"])
+    assert type_info == "tensor<1x3x224x224xf32>"
+
+
+def test_visit_value_info_multiple_time():
```

Comment:
```suggestion
def test_visit_value_info():
```
Let's remove the multiple_time from the name, it's makes it sound like it visits it multiple times.

Reply:
I mean it doesn't only test that, it checks the behaviour, including what happens when it visits it multiple times.

Reply:
Ah, let's merge them